packages/pouch-storage/src/PouchCommandHistoryTable.ts

```diff
@@ -3,26 +3,258 @@ import {
-            ]),
-          ]
-        : []
+  constructor(language: string, private cache: PouchCommandHistoryCache) {
+    super(`CommandHistoryStorage.${language}`, ({
+      // Optimizations to cut down on growing table size. These should be safe
+      // since we don't care about revision history for command history
+      // documents.
+      auto_compaction: true,
+      revs_limit: 1,
+    } as unknown) as PouchDB.HttpAdapter.HttpAdapterConfiguration);
+
+    // Add this table instance to the table registry
+    if (!this.cache.tableRegistry.has(this.cacheKey)) {
+      this.cache.tableRegistry.set(this.cacheKey, new Set());
+    }
+
+    log.debug('Adding table to registry', this.cacheKey);
+    this.cache.tableRegistry.get(this.cacheKey)?.add(this);
+  }
+
+  private searchText?: string;
+
+  private get cacheKey(): string {
+    return this.db.name;
+  }
+
+  setSearchDebounceTimeout?: number;
+
+  setSearch(searchText: string): void {
+    clearTimeout(this.setSearchDebounceTimeout);
+
+    // Debounced search to minimize querying the PouchDB db while user types
+    this.setSearchDebounceTimeout = window.setTimeout(() => {
+      log.debug('Setting search filters', searchText);
+
+      this.searchText = searchText.trim().toLowerCase();
+
+      this.setFilters(
+        searchText
+          ? [
+              StorageUtils.makeFilterConfig([
+                StorageUtils.makeFilterItem(
+                  'name',
+                  FilterType.contains,
+                  searchText
+                ),
+              ]),
+            ]
+          : []
+      );
+    }, 500);
+  }
+
+  // Our current version of eslint + prettier doesn't always like the `override`
+  // keyword. Whenever we upgrade, we should annotate this function with `override`.
+  dbUpdate(
+    event: PouchDB.Core.ChangesResponseChange<CommandHistoryStorageItem>
+  ): void {
+    log.debug('Clearing cache and refreshing data', event);
+
+    this.cache.response.delete(this.cacheKey);
+
+    super.dbUpdate(event);
+  }
+
+  /**
+   * Fetch command history data from `this.cache.response` or from
+   * PouchDB if data is not found in the cache. If the number of total items in
+   * the db exceeds COMMAND_HISTORY_ITEMS_MAX, the database will be pruned down
+   * to COMMAND_HISTORY_ITEMS_PRUNE total items. Note that PouchDB doesn't
+   * actually remove them from the underlying IndexDB, they are just marked
+   * as deleted and won't be present in our query results.
+   * @param selector
+   */
+  private async fetchData(
+    _selector: PouchDB.Find.Selector
+  ): Promise<
+    PouchDB.Find.FindResponse<CommandHistoryStorageItem & PouchStorageItem>
+  > {
+    // Wrapping this in a setTimeout so that it executes on next call stack.
+    // This is necessary to ensure `this.cache` has been initialized due to some
+    // nuances with property initilization with inherited classes.
+    return PromiseUtils.withTimeout(0, async () => {
```

Comment:
I don't think this needs to be wrapped in a timeout? What about cache isn't initialized at this point?

Reply:
Ahh yea that's right. Cool.